Source/SwiftLintFramework/Rules/Performance/EmptyCountRule.swift

```diff
@@ -33,20 +35,96 @@ public struct EmptyCountRule: ConfigurationProviderRule, OptInRule {
         ]
     )
 
-    public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        let defaultPattern = #"\bcount\s*(==|!=|<|<=|>|>=)\s*0(\b|([box][0_]+\b){1})"#
-        let prefixPattern = configuration.onlyAfterDot ? #"\."# : ""
-        let pattern = prefixPattern + defaultPattern
+    public func preprocess(syntaxTree: SourceFileSyntax) -> SourceFileSyntax? {
+        syntaxTree.folded()
+    }
+
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor {
+        Visitor(onlyAfterDot: configuration.onlyAfterDot)
+    }
+}
+
+private extension EmptyCountRule {
+    final class Visitor: ViolationsSyntaxVisitor {
+        private let onlyAfterDot: Bool
+
+        init(onlyAfterDot: Bool) {
+            self.onlyAfterDot = onlyAfterDot
+            super.init(viewMode: .sourceAccurate)
+        }
+
+        private let operators: Set = ["==", "!=", ">", ">=", "<="]
```

Comment:
I guess I need to add `<` back here to support both directions (Yoda conditions)